TFNetworkLayer.py

```diff
@@ -840,12 +844,19 @@ def batch_norm(self, data,
       https://github.com/deepmind/sonnet/blob/master/sonnet/python/modules/batch_norm.py
     """
     with tf.variable_scope("batch_norm"):
-      x = data.get_placeholder_flattened(keep_dims=True)  # shape (time',...)
-      mean, variance = tf.nn.moments(x, axes=[0], keep_dims=True)
+      if masked_time:
+        keep_dims = False if fused else True
+        x = data.get_placeholder_flattened(keep_dims=keep_dims)
+        mean, variance = tf.nn.moments(x, axes=[0], keep_dims=keep_dims)
+        shape = [data.dim] if fused else data.get_bc_spatial_batch_shape()
```

Comment:
`shape` looks very much wrong to me. If `fused`, you collapse batch + time together into a single, but then you still have 2 axes (at least), e.g. (batch*time, dim).

Reply:
But as shape should match for broadcasting, it should be like `(1, data.dim)` or so. Actually better just use `list(data.get_bc_spatial_batch_shape())` and then remove the time-axis.

Reply:
Ah sorry, I think I was wrong about `shape`. But the code is slightly confusing. Make it depend on `keep_dims`, not on `fused`.
(And also consider / keep in mind that at some point, maybe it's actually more clean to have `keep_dims = False` in all case, so maybe we want to change the code later.)